### ISSUE TITLE
AKSv2 follow ups

### ISSUE THREAD

**Issue by cbron** [rancher/rancher#31648]
1. `createCASecret` should be revamped if possible - https://github.com/rancher/aks-operator/pull/1#discussion_r587341893
2. The cluster should go into an updating phase (and that should be shown to user) when it is edited - https://github.com/rancher/aks-operator/pull/1#discussion_r566827480
3. `updateUpstreamClusterState` needs nil checks to respect upstream - https://github.com/rancher/aks-operator/pull/1#discussion_r564929005
4. Similar to part 3, nils should only be checked, not initialized to respect upstream : https://github.com/rancher/aks-operator/blob/a937363f350472fdd1b7b8e913c56eb8a8c327e1/controller/aks-cluster-config-handler.go#L502-L504

In short: 
> don't evaluate nil fields when updating, don't overwrite the nil value on spec unless user specified, don't use nil value to indicate default should be used

**Comment by mjura** [comment]
Regarding solutions:
1. We have to keep `createCASecret` because it is the easiest way to get `CA` and create secretToken
2. Was not addressed yet
3. Was done in https://github.com/rancher/aks-operator/pull/3/
4. Was done in https://github.com/rancher/aks-operator/pull/3/

**Comment by mjura** [comment]
Follow ups from AKSv2 PR3 https://github.com/rancher/aks-operator/pull/3/

- Checking the node pools for updates first means we can't update the cluster kubernetes version and a node pool kubernetes version at the same time. If the cluster version was updated first then the node pool versions could follow and there wouldn't be an error. https://github.com/rancher/aks-operator/pull/3#discussion_r616239775
- Something to think about for a different PR, but I was having trouble deleting clusters from Rancher because the cluster object gets deleted immediately but the aksclusterconfig object blocks while being deleted here. Maybe the controller could set a "deleting" state that the cluster handler can check before deleting? https://github.com/rancher/aks-operator/pull/3#discussion_r622436954

**Comment by thedadams** [comment]
@Caleb, I think this is good:
1. It seems to be the way to get the endpoint and CA from AKS
2. This was addressed by @mjura
3. Addressed as above
4. Addressed as above
